cds_dojson/marc21/fields/books/book.py

```diff
@@ -0,0 +1,662 @@
+        if not empty and i < len(_publication_info):
+            _publication_info[i].update(temp_info)
+        else:
+            _publication_info.append(temp_info)
+
+    return _publication_info
+
+
+@model.over('related_records', '(^775__)|(^787__)')
+@filter_list_values
+@for_each_value
+def related_records(self, key, value):
+    """Translates related_records field."""
+    try:
+        if key == '775__':
+            clean_val('b', value, str, manual=True)
+            clean_val('c', value, str, manual=True)
+        if key == '787__':
+            clean_val('i', value, str, manual=True)
+    except ManualMigrationRequired as e:
+        # TODO logs
+        raise e
+    return {'record': {'$ref': rel_url(clean_val('w', value, str, req=True))}}
+
+
+@model.over('accelerator_experiments', '^693__')
+@filter_list_values
+@for_each_value
+def accelerator_experiments(self, key, value):
+    """Translates accelerator_experiments field."""
+    return {'accelerator': clean_val('a', value, str),
+            'experiment': clean_val('e', value, str),
+            }
+
+
+# TODO - discuss how we would like to keep links to holdings (files and ebooks)
+# TODO maybe regex for links?
+@model.over('urls', '^8564_')
+@for_each_value
+@filter_values
+def urls(self, key, value):
+    """Translates urls field."""
+    try:
+        clean_val('y', value, str, manual=True)
+    except ManualMigrationRequired as e:
+        raise e
+    return {'value': clean_val('u', value, str, req=True)}
+
+
+@model.over('isbns', '^020__')
+@filter_list_values
+def isbns(self, key, value):
+    """Translates isbns fields."""
+    _isbns = self.get('isbns', [])
+    for v in force_list(value):
+        subfield_u = clean_val('u', v, str)
+        isbn = {'value': clean_val('a', v, str) or clean_val('z', v, str)}
+        if not isbn['value']:
+            raise ManualMigrationRequired
+        if subfield_u:
+            volume = re.search(r'(\(*v[.| ]*\d+.*\)*)', subfield_u)
+
+            if volume:
+                volume = volume.group(1)
+                subfield_u = subfield_u.replace(volume, '').strip()
+                existing_volume = self.get('volume')
+                if existing_volume:
+                    raise ManualMigrationRequired
+                self['volume'] = volume
+            if subfield_u.upper() in MEDIUM_TYPES:
+                isbn.update({'medium': subfield_u})
+            else:
+                isbn.update({'description': subfield_u})
+        # TODO subfield C
+        if isbn not in _isbns:
+            _isbns.append(isbn)
+    return _isbns
+
+
+@model.over('standard_numbers', '^021__')
+@for_each_value
+@filter_values
+def standard_numbers(self, key, value):
+    """Translates standard numbers values."""
+    a = clean_val('a', value, str)
+    b = clean_val('b', value, str)
+    sn = a or b
+    if sn:
+        return {'value': sn,
+                'hidden': True if b else None}
```

Comment:
`else False` ?

Reply:
It's with `None` because the `@filter_values` decorator filters only None from the output list. In that case field hidden just does not exist but it's better to be set as False as you say - I will fix it